core/modules/replica/Controller.h

```diff
@@ -27,22 +27,174 @@
  */
 
 // System headers
+#include <functional>
 #include <map>
 #include <memory>
 #include <thread>
 #include <vector>
 
 // Qserv headers
 #include "replica/Request.h"
-#include "replica/RequestTypesFwd.h"
 #include "replica/ServiceProvider.h"
 #include "util/Mutex.h"
 
 // Forward declarations
 namespace lsst {
 namespace qserv {
 namespace replica {
-    class ControllerImpl;
+
+    class ReplicationRequest;
+    class DeleteRequest;
+
+    typedef std::shared_ptr<ReplicationRequest> ReplicationRequestPtr;
+    typedef std::shared_ptr<DeleteRequest>      DeleteRequestPtr;
+
+    class FindRequest;
+    class FindAllRequest;
+
+    typedef std::shared_ptr<FindRequest>    FindRequestPtr;
+    typedef std::shared_ptr<FindAllRequest> FindAllRequestPtr;
+
+    class EchoRequest;
+
+    typedef std::shared_ptr<EchoRequest> EchoRequestPtr;
+
+    class IndexRequest;
```

Comment:
nit: I don't understand why these comein groups of ones and twos.

Reply:
This is just due to historical reasons. I've rearranged type definitions by regrouping them.